### PR TITLE
Add home page map to search for local businesses

### PR DESCRIPTION
**Pull request by winniewan**
- The home page will have a search box along with a map underneath. 
- User can use the search box to enter a location they wish to view business nearby. There will be a autocomplete tool from the Google Maps API. 
- After the user has entered a location, the map will display marker points along with a results table to show the results from the search. 
- User can interact with either the map marker or the results table to navigate to their interested nearby business profiles.

### CODE REVIEW

File: capstone/src/main/webapp/js/index.js [hunk 1]
```diff
@@ -79,5 +86,141 @@ function showMoreInfo(id) {
...
+  tr.onclick = function() {
+    window.location.assign('/business.html?id='+id);
+  };
+
+  let iconTd = document.createElement('td');
+  let nameTd = document.createElement('td');
+  let icon = document.createElement('img');
+  icon.src = markerIcon;
+  icon.setAttribute('class', 'placeIcon');
+  icon.setAttribute('className', 'placeIcon');
```
Comment (eash130) [inline capstone/src/main/webapp/js/index.js:197]: Why are you doing this twice? I'm not sure if the second one works/is even needed.
You could just do icon.className = 'placeIcon'

File: capstone/src/main/webapp/css/index.css [hunk 2]
```diff
@@ -1,39 +1,106 @@
...
+    font-weight: bold;
+    text-align: left;
+}
+
+#listing {
+    width: 200px;
+    height: 470px;
+    overflow: auto;
+    cursor: pointer;
+    overflow-x: hidden;
```
Comment (eash130) [inline capstone/src/main/webapp/css/index.css:59]: nit: alphabetize the properties for all.

File: capstone/src/main/webapp/js/index.js [hunk 3]
```diff
@@ -79,5 +86,141 @@ function showMoreInfo(id) {
...
+  let marker = this;
+  infoWindow.open(map, marker);
+  buildIWContent(id, name, location);
+}
+
+// Load the place information into the HTML elements used by the info window.
+function buildIWContent(id, name, location) {
+  document.getElementById('map-name').textContent = name;
+  document.getElementById('map-location').textContent = location;
+  document.getElementById('map-link').setAttribute('href', '/business.html?id='+id);
```
Comment (eash130) [inline capstone/src/main/webapp/js/index.js:225]: you can just do
document.getElementById('map-link').href = '/business.html?id=' + id

just as another option.

File: capstone/src/main/webapp/css/index.css (hunk 2 above)
Comment (winniewan) [reply capstone/src/main/webapp/css/index.css]: done

File: capstone/src/main/webapp/js/index.js [hunk 4]
```diff
@@ -79,5 +86,141 @@ function showMoreInfo(id) {
...
+  autocomplete.addListener('place_changed', onPlaceChanged);
+}
+
+// When the user enters a location, get the place details for the location and
+// zoom the map in on the location.
+function onPlaceChanged() {
+  let place = autocomplete.getPlace();
+  if (place.geometry) {
+    map.panTo(place.geometry.location);
+    // map.setZoom(10);
```
[the remark this answers is not in the document]
Comment (winniewan) [reply capstone/src/main/webapp/js/index.js]: done

File: capstone/src/main/webapp/js/index.js [hunk 5]
```diff
@@ -79,5 +86,141 @@ function showMoreInfo(id) {
...
+  let bounds = map.getBounds();
+  let NEPoint = bounds.getNorthEast();
+  let SWPoint = bounds.getSouthWest();
+
+  let SW_Lat = SWPoint.lat();
+  let SW_Lng = SWPoint.lng();
+  let NE_Lat = NEPoint.lat();
+  let NE_Lng = NEPoint.lng();
+
+  fetch('/map?SW_Lat='+SW_Lat+'&SW_Lng='+SW_Lng+'&NE_Lat='+NE_Lat+'&NE_Lng='+NE_Lng).then(response => response.json()).then(results => {
```
[the remark this answers is not in the document]
Comment (winniewan) [reply capstone/src/main/webapp/js/index.js]: done

File: capstone/src/main/webapp/js/index.js (hunk 1 above)
Comment (winniewan) [reply capstone/src/main/webapp/js/index.js]: good question! thanks for catching it, removed.

File: capstone/src/main/webapp/js/index.js (hunk 3 above)
Comment (winniewan) [reply capstone/src/main/webapp/js/index.js]: done

**Pull request merged**